kernels/available/python/default.nix

```diff
@@ -12,29 +12,52 @@
   projectDir ? self + "/kernels/available/python",
   pyproject ? projectDir + "/pyproject.toml",
   poetrylock ? projectDir + "/poetry.lock",
-  overrides ? poetry2nix.overrides.withDefaults (import ./overrides.nix),
+  overrides ? import ./overrides.nix,
```

Comment:
Why move `poetry2nix.overrides.withDefaults` in the function body?

Reply:
I was trying to do too many things with this PR. The reason to move the `poetry2nix` call to the function body is because it makes it easier to simply pass an `overrides` function when the user just wants to use the provided `poetry2nix`.

I'm reverting this back for this PR and I'll start commenting about my UX concerns on a separate issue. I already got some illustrative examples to kick off the discussion.